### ISSUE TITLE
Loading time at launching is much more than the previous version... lately the app when launched it doesn't remember my active provider and  picks another randomly

### ISSUE THREAD

**Issue by SANCTI-afk** [recloudstream/cloudstream#55]
### Steps to reproduce

.

### Expected behavior

.

### Actual behavior

.

### Cloudstream version and commit hash

3.1.3

### Android version

9

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to pre-release version **[Latest](https://github.com/recloudstream/cloudstream/releases)**.
- [X] If related to a provider, I have checked the site and it works, but not the app.
- [X] I will fill out all of the requested information in this form.

**Comment by C10udburst** [comment]
the speed issue is caused most likely by your slow internet connection. the app has to download multiple JSON files on start in order to check if any extensions need updating. you can try to disable autoupdate or [install extensions locally](https://recloudstream.github.io/users/local-plugins.html)

**Comment by C10udburst** [comment]
also try after 0f8a2df9e5046c5f1822f19e85dd2248e6b1737b

**Comment by Johnny5sword** [comment]
I also have an issue related to the 2nd issue in the title.

### Steps to reproduce
1. Open app
2. Tap provider selection button in lower right corner of the home tab
3. Pick any provider aside from none and random
4. Change provider to none
5. Change tab in app (e.g. tap search tab) and go back to home tab 
6. Or after step 4, close app and open app
   
### Expected behavior
In the home tab, after changing from another provider to 'none', that choice persists even after changing tabs or closing the app.
### Actual behavior
The provider reverts back to the previous provider after selecting none in the events of changing tabs or reopening the app.
### Cloudstream version and commit hash
3.1.3 (e503504)
### Android version
11 (LineageOS 18.1)
### Other details
My workaround is to reinstall the app and avoid changing from my active provider which is none in the first place.

@SANCTI-afk , also try changing the home tab active provider to none if you haven't already. It may make the startup a bit faster.

**Comment by Johnny5sword** [comment]
I can no longer replicate my reported issue. The bug is fixed on my side after updating to 3.1.5 (6f60298)
